togetherly/lib/views/widgets/new_chore_dialog.dart

```diff
@@ -0,0 +1,400 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:togetherly/models/child.dart';
+import 'package:togetherly/models/chore.dart';
+import 'package:togetherly/models/person.dart';
+import 'package:togetherly/providers/chore_provider.dart';
+import 'package:togetherly/themes.dart';
+
+class NewChoreDialog extends StatefulWidget {
+  const NewChoreDialog({super.key, this.assignedChildId});
+  final int? assignedChildId;
+
+  @override
+  State<NewChoreDialog> createState() => _NewChoreDialogState();
+}
+
+class _NewChoreDialogState extends State<NewChoreDialog> {
+  final _formKey = GlobalKey<FormState>();
+  final _titleController = TextEditingController();
+  final _descriptionController = TextEditingController();
+  final _pointsController = TextEditingController();
+  DateTime _dueDate = DateTime.now();
+  String _title = '';
+  String _description = '';
+  int _points = 0;
+  bool _isBonus = false;
+  bool _isShared = false;
+  bool _loading = false;
+  final List<Map<String, dynamic>> _repeatingWeekdays = [
+    {
+      'day': 'S',
+      'index': 0,
+      'selected': false,
+    },
+    {
+      'day': 'M',
+      'index': 1,
+      'selected': false,
+    },
+    {
+      'day': 'T',
+      'index': 2,
+      'selected': false,
+    },
+    {
+      'day': 'W',
+      'index': 3,
+      'selected': false,
+    },
+    {
+      'day': 'Th',
+      'index': 4,
+      'selected': false,
+    },
+    {
+      'day': 'F',
+      'index': 5,
+      'selected': false,
+    },
+    {
+      'day': 'Sa',
+      'index': 6,
+      'selected': false,
+    },
+  ];
+  final List<Child> _assignedPeople = [];
+
+  List<Child> peopleList = const [
+    Child(
+      familyId: 0,
+      name: 'Emma',
+      icon: ProfileIcon.bear,
+      totalPoints: 45,
+    ),
+    Child(
+      familyId: 0,
+      name: 'Jacob',
+      icon: ProfileIcon.dog,
+      totalPoints: 80,
+    ),
+    Child(
+      familyId: 0,
+      name: 'Natalie',
+      icon: ProfileIcon.cat,
+      totalPoints: 65,
+    ),
+    Child(
+      familyId: 0,
+      name: 'Robert',
+      icon: ProfileIcon.giraffe,
+      totalPoints: 30,
+    ),
+  ];
+
+  void submitChore(BuildContext context, ChoreProvider provider) {
+    if (_formKey.currentState!.validate()) {
+      final newChore = Chore(
+        title: _title,
+        description: _description,
+        points: _points,
+        dueDate: _dueDate,
+        isShared: _isShared,
+        assignedChildId: 1,
+      );
+      setState(() => _loading = true);
+      provider.addChore(newChore);
+      setState(() => _loading = false);
```

Comment:
This should not occur until the future returned by addChore has returned. You could make this function async, or you could use the `Future.then` method ([see docs](https://dart.dev/guides/libraries/futures-error-handling#the-future-api-and-callbacks)).

Reply:
Will fix